### ISSUE TITLE
run.py won't work if it has '!' in the RHN password

### ISSUE THREAD

**Issue by alberttwong** [2015-Middleware-Keynote/demo-ansible#141]
I had to change my password to not include the "!" character.

**Comment by alberttwong** [comment]
Didn't work if I put the password in "" or ''

**Comment by thoraxe** [comment]
@detiber this seems like either a python issue or a bash issue

**Comment by douglasawoods** [comment]
@Albert If this is rhsm-pass, try using no quotes whatsoever.

**Comment by detiber** [comment]
@thoraxe indeed, the value is serialized between python strings, yaml strings and then fed into bash, so some type of a quoting issue is at play. It's just a matter of tracking it down.

**Comment by thoraxe** [comment]
I'll leave this issue open for @alberttwong but there's nothing actually "wrong" in the scripts. It's just a matter of escaping correctly, unfortunately.

You have three layers of interpolation:

1) your command entered into bash
2) how python munges it
3) how Ansible then munges that
